Assets/BossRoom/Scripts/Shared/Net/UnityServices/Auth/SubIdentity_Authentication.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using Unity.Services.Authentication;
+
+namespace BossRoom.Scripts.Shared.Net.UnityServices.Auth
+{
+    /// <summary>
+    /// The Authentication package will sign in asynchronously and anonymously. When complete, we will need to store the generated ID.
+    /// </summary>
+    public class SubIdentity_Authentication : SubIdentity, IDisposable
+    {
+        private bool m_hasDisposed = false;
+
+        /// <summary>
+        /// This will kick off a login.
+        /// </summary>
+        public SubIdentity_Authentication(Action onSigninComplete = null)
+        {
+            DoSignIn(onSigninComplete);
+        }
+        ~SubIdentity_Authentication()
+        {
+            Dispose();
+        }
+        public void Dispose()
+        {
+            if (!m_hasDisposed)
+            {
+                AuthenticationService.Instance.SignedIn -= OnSignInChange;
+                AuthenticationService.Instance.SignedOut -= OnSignInChange;
+                m_hasDisposed = true;
+            }
+        }
+
+        private async void DoSignIn(Action onSigninComplete)
+        {
+            await Unity.Services.Core.UnityServices.InitializeAsync();
+            AuthenticationService.Instance.SignedIn += OnSignInChange;
+            AuthenticationService.Instance.SignedOut += OnSignInChange;
+
+            try
+            {   if (!AuthenticationService.Instance.IsSignedIn)
+                    await AuthenticationService.Instance.SignInAnonymouslyAsync(); // Don't sign out later, since that changes the anonymous token, which would prevent the player from exiting lobbies they're already in.
+                onSigninComplete?.Invoke();
+            }
+            catch
+            {   UnityEngine.Debug.LogError("Failed to login. Did you remember to set your Project ID under Services > General Settings?");
+                throw;
+            }
+
+            // Note: If for some reason your login state gets weird, you can comment out the previous block and instead call AuthenticationService.Instance.SignOut().
+            // Then, running Play mode will fail to actually function and instead will log out of your previous anonymous account.
+            // When you revert that change and run Play mode again, you should be logged in as a new anonymous account with a new default name.
+        }
+
+        private void OnSignInChange()
+        {
+            SetContent("id", AuthenticationService.Instance.PlayerId);

```

Comment:
This wrapper class appears to be setup to manage a statically accessible property.  Can we simplify it?

Reply:
The reemergence of that class was caused by a git user error, it's actually has been removed from the project.

Reply:
It also relates to some of other conversations that I resolved - I'll mark them accordingly with a -1 emoji just so that it's readable later.